langkit/support/langkit_support-symbols.adb

```diff
@@ -75,11 +75,19 @@ package body Langkit_Support.Symbols is
      (ST : Symbol_Table; Index : Precomputed_Symbol_Index) return Symbol_Type
    is
    begin
-      --  For languages that carry no precomputed symbols, Index can have no
-      --  value, so we have noisy but useless warning.
       pragma Warnings (Off, "value not in range");
+      return Get_Symbol (ST, Precomputed_Symbol (ST, Index));
+   end Precomputed_Symbol;
+
+   ------------------------
+   -- Precomputed_Symbol --
+   ------------------------
+
+   function Precomputed_Symbol
+     (ST : Symbol_Table; Index : Precomputed_Symbol_Index) return Thin_Symbol
+   is
+   begin
       return ST.Precomputed (Index);
-      pragma Warnings (On, "value not in range");
```

Comment:
Please restore the comment that explains why the `pragma Warnings (Off)` is needed. Also, why have you removed the `pragma Warnings (On)` part? I think it should stay.